### DISCUSSION TITLE
Feedback widget with navigation.instant

### DISCUSSION THREAD

**Discussion by wandri** [squidfunk/mkdocs-material#4258] [Q&A] [answered]
Hi, thanks for the work on this repository, it's awesome.

I'm strugling to find a nice way to implement a custom feedback widget with the `navigation.instant` parameter. Because in my implementation, it's displayed in the first page but after a navigation, it's not anymore displayed.

In the [google analytics script](https://github.com/squidfunk/mkdocs-material/blob/612429c903b1edee68a5fcd561e7b5f3b54ec362/src/partials/integrations/analytics/google.html#L51-L74), 
* It uses `DOMContentLoaded` but it's only invoked once at the beginning.
* It uses `location$.subscribe` but it is invoked before the load of the page in XHR.

Currently I'm doing a hack wich is:

```
if (typeof location$ !== "undefined") {
  location$.subscribe(function () {
    setTimeout(() => {
      generateFeedback()
    }, 400)
  })
}
 ```

But it's an hack. And if the XHR call makes more than 400ms, it won't work.

I'm probably missing something. **Do you know if there is a nicer and better way ?**

**Comment by squidfunk** [comment, marked as answer]
The `location$` emits as soon as navigation is initiated. You probably want to wait for the document to be loaded:

``` js
document$.subscribe(() => {
  console.log("page loaded via instant loading")
})
```

**Reply by wandri** [reply]
Thanks @squidfunk , it works well!
